### ISSUE TITLE
Linux: Process exits right away after clean installation

### ISSUE THREAD

**Issue by ahmetalpbalkan** [Azure/azure-xplat-cli#2163]
I just installed a clean "Ubuntu 14.04.3 LTS" VM and installed the CLI. However CLI process exits right away with exitcode=1 without any output at all, it is unusable.

```
$ sudo apt-get upgrade
$ sudo apt-get -qqy install npm
$ sudo npm install -g azure-cli
$ exit
[logout/login]
$ azure; echo ExitCode=$?
1 <-------------- just exited without any output. ExitCode=1
```

### strace output:

```
{23:24}~ ➭ strace azure
execve("/usr/local/bin/azure", ["azure"], [/* 25 vars */]) = 0
brk(0)                                  = 0x89b000
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f7c80577000
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
open("/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=25750, ...}) = 0
mmap(NULL, 25750, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f7c80570000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\320\37\2\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=1840928, ...}) = 0
mmap(NULL, 3949248, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f7c7ff92000
mprotect(0x7f7c8014d000, 2093056, PROT_NONE) = 0
mmap(0x7f7c8034c000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1ba000) = 0x7f7c8034c000
mmap(0x7f7c80352000, 17088, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f7c80352000
close(3)                                = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f7c8056f000
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f7c8056d000
arch_prctl(ARCH_SET_FS, 0x7f7c8056d740) = 0
mprotect(0x7f7c8034c000, 16384, PROT_READ) = 0
mprotect(0x605000, 4096, PROT_READ)     = 0
mprotect(0x7f7c80579000, 4096, PROT_READ) = 0
munmap(0x7f7c80570000, 25750)           = 0
brk(0)                                  = 0x89b000
brk(0x8bc000)                           = 0x8bc000
open("/usr/lib/locale/locale-archive", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=1607664, ...}) = 0
mmap(NULL, 1607664, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f7c803e4000
close(3)                                = 0
execve("/usr/local/sbin/node", ["node", "/usr/local/bin/azure"], [/* 25 vars */]) = -1 ENOENT (No such file or directory)
execve("/usr/local/bin/node", ["node", "/usr/local/bin/azure"], [/* 25 vars */]) = -1 ENOENT (No such file or directory)
execve("/usr/sbin/node", ["node", "/usr/local/bin/azure"], [/* 25 vars */]) = 0
brk(0)                                  = 0x255f000
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f1e33daf000
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
open("/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=25750, ...}) = 0
mmap(NULL, 25750, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f1e33da8000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/usr/lib/libax25.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\340%\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=43624, ...}) = 0
mmap(NULL, 2139632, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f1e33984000
mprotect(0x7f1e3398d000, 2097152, PROT_NONE) = 0
mmap(0x7f1e33b8d000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x9000) = 0x7f1e33b8d000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/usr/lib/libax25io.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0P\r\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=14392, ...}) = 0
mmap(NULL, 2113768, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f1e3377f000
mprotect(0x7f1e33781000, 2097152, PROT_NONE) = 0
mmap(0x7f1e33981000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x2000) = 0x7f1e33981000
mmap(0x7f1e33983000, 232, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f1e33983000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\320\37\2\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=1840928, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f1e33da7000
mmap(NULL, 3949248, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f1e333ba000
mprotect(0x7f1e33575000, 2093056, PROT_NONE) = 0
mmap(0x7f1e33774000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1ba000) = 0x7f1e33774000
mmap(0x7f1e3377a000, 17088, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f1e3377a000
close(3)                                = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f1e33da5000
arch_prctl(ARCH_SET_FS, 0x7f1e33da5740) = 0
mprotect(0x7f1e33774000, 16384, PROT_READ) = 0
mprotect(0x7f1e33981000, 4096, PROT_READ) = 0
mprotect(0x7f1e33b8d000, 4096, PROT_READ) = 0
mprotect(0x60d000, 4096, PROT_READ)     = 0
mprotect(0x7f1e33db1000, 4096, PROT_READ) = 0
munmap(0x7f1e33da8000, 25750)           = 0
rt_sigaction(SIGALRM, {0x404db0, [ALRM], SA_RESTORER|SA_RESTART, 0x7f1e333f0d40}, {SIG_DFL, [], 0}, 8) = 0
rt_sigaction(SIGTERM, {0x404db0, [TERM], SA_RESTORER|SA_RESTART, 0x7f1e333f0d40}, {SIG_DFL, [], 0}, 8) = 0
rt_sigaction(SIGINT, {0x404db0, [INT], SA_RESTORER|SA_RESTART, 0x7f1e333f0d40}, {SIG_DFL, [], 0}, 8) = 0
rt_sigaction(SIGQUIT, {0x404db0, [QUIT], SA_RESTORER|SA_RESTART, 0x7f1e333f0d40}, {SIG_DFL, [], 0}, 8) = 0
rt_sigaction(SIGSEGV, {0x404db0, [SEGV], SA_RESTORER|SA_RESTART, 0x7f1e333f0d40}, {SIG_DFL, [], 0}, 8) = 0
rt_sigaction(SIGPIPE, {SIG_IGN, [PIPE], SA_RESTORER|SA_RESTART, 0x7f1e333f0d40}, {SIG_DFL, [], 0}, 8) = 0
socket(PF_LOCAL, SOCK_DGRAM, 0)         = 3
brk(0)                                  = 0x255f000
brk(0x2580000)                          = 0x2580000
open("/proc/net/dev", O_RDONLY)         = 4
fstat(4, {st_mode=S_IFREG|0444, st_size=0, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f1e33dae000
read(4, "Inter-|   Receive               "..., 1024) = 572
ioctl(3, SIOCGIFHWADDR, {ifr_name="eth0", ifr_hwaddr=00:0d:3a:33:1e:f9}) = 0
ioctl(3, SIOCGIFHWADDR, {ifr_name="docker0", ifr_hwaddr=02:42:19:3a:b3:f9}) = 0
ioctl(3, SIOCGIFHWADDR, {ifr_name="lo", ifr_hwaddr=00:00:00:00:00:00}) = 0
read(4, "", 1024)                       = 0
close(4)                                = 0
munmap(0x7f1e33dae000, 4096)            = 0
open("/etc/ax25/axports", O_RDONLY)     = 4
fstat(4, {st_mode=S_IFREG|0644, st_size=200, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f1e33dae000
read(4, "# /etc/ax25/axports\n#\n# The form"..., 4096) = 200
read(4, "", 4096)                       = 0
close(3)                                = 0
close(4)                                = 0
munmap(0x7f1e33dae000, 4096)            = 0
open("/etc/localtime", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=118, ...}) = 0
fstat(3, {st_mode=S_IFREG|0644, st_size=118, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f1e33dae000
read(3, "TZif2\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\1\0\0\0\1\0\0\0\0"..., 4096) = 118
lseek(3, -62, SEEK_CUR)                 = 56
read(3, "TZif2\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\1\0\0\0\1\0\0\0\0"..., 4096) = 62
close(3)                                = 0
munmap(0x7f1e33dae000, 4096)            = 0
getpid()                                = 42406
socket(PF_LOCAL, SOCK_DGRAM|SOCK_CLOEXEC, 0) = 3
connect(3, {sa_family=AF_LOCAL, sun_path="/dev/log"}, 110) = 0
sendto(3, "<187>Sep 30 23:24:45 node[42406]"..., 63, MSG_NOSIGNAL, NULL, 0) = 63
exit_group(1)                           = ?
+++ exited with 1 +++
```

any ideas what's going on here?

**Comment by yugangw-msft** [comment]
@ahmetalpbalkan 
i don't see this in my "UBuntu 15.04", i can try your configuration and see whether this reproes.
At the same time, could you check out what version node you are using? I would like understand more the following line of the trace
    execve("/usr/local/sbin/node", ["node", "/usr/local/bin/azure"], [/* 25 vars */]) = -1 ENOENT (No such file or directory)

**Comment by ahmetalpbalkan** [comment]
@yugangw-msft I just purged the nodejs package and retried. Turns it it was a corrupt installation, it's working now... However, when I used Ubuntu's PPA and installed `nodejs` and then `npm` packages using `apt-get` and then `sudo npm install -g azure-cli`, running `azure` command gave me the following:

```
$ azure
/usr/bin/env: node: No such file or directory
```

so I had to symlink it (`sudo ln -s /usr/bin/nodejs /usr/bin/node`) and then it worked... Could someone else besides me check on Ubuntu 14.04.3 apt repos, is this the case in clean installation? I'm no nodejs expert at all, I might be wrong but it seemed like there's this particular friction in there. 

Feel free to close the issue if it's just me.
